src/Particular.ThroughputCollector.Persistence.InMemory/InMemoryThroughputDataStore.cs

```diff
@@ -2,57 +2,66 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using Contracts;
 
-class InMemoryThroughputDataStore : IThroughputDataStore
+class InMemoryThroughputDataStore(PersistenceSettings persistenceSettings) : IThroughputDataStore
 {
-    private readonly List<Endpoint> endpoints;
-    private List<BrokerData> brokerData;
+    private readonly EndpointCollection endpoints = [];
+    private readonly List<BrokerData> brokerData = [];
 
-    public InMemoryThroughputDataStore()
+    public Task<IEnumerable<Endpoint>> GetAllEndpoints(bool includePlatformEndpoints = true, CancellationToken cancellationToken = default)
```

Comment:
Should this return `IAsyncEnumerable`?

Reply:
https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/querying/how-to-stream-query-results

Reply:
Lol thanks.  I am aware of the interface, but it kept throwing exceptions when I tried to use it.  I will be creating a minimal repro and raising an issue with RavenDB.